### ISSUE TITLE
Can't create read replica using terraform resource azurerm_cosmosdb_postgresql_cluster

### ISSUE THREAD

**Issue by leilifremont** [hashicorp/terraform-provider-azurerm#22641]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.2.9

### AzureRM Provider Version

3.63.0

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_postgresql_cluster

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    local = {
      source = "hashicorp/local"
    }
    azurerm = {
      source = "hashicorp/azurerm"

      version = "= 3.63.0"
    }
    random = {
      source  = "hashicorp/random"
      version = "~> 3.4.3"
    }
    azapi = {
      source  = "azure/azapi"
      version = ">= 0.6.0"
    }
    azuread = {
      source  = "hashicorp/azuread"
      version = "~> 2.15.0"
    }
    ec = {
      source  = "elastic/ec"
      version = "= 0.7.0"
    }
    external = {
      source  = "hashicorp/external"
      version = "= 2.3.1"
    }
    elasticstack = {
      source  = "elastic/elasticstack"
      version = "= 0.6.1"
    }
  }

  required_version = ">= 1.0"
}
```

### Debug Output/Panic Output

```shell
Various errors:

1.
╷
│ Error: Missing required argument
│
│   with azurerm_cosmosdb_postgresql_cluster.read_replica,
│   on main.tf line 34, in resource "azurerm_cosmosdb_postgresql_cluster" "read_replica":
│   34:   source_resource_id              = azurerm_cosmosdb_postgresql_cluster.primary_replica.id
│
│ "source_resource_id": all of `point_in_time_in_utc,source_location,source_resource_id` must be specified
╵

2.
╷
│ Error: creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request" Message="The source cluster is not ready for restore yet."
│
│   with azurerm_cosmosdb_postgresql_cluster.read_replica,
│   on main.tf line 23, in resource "azurerm_cosmosdb_postgresql_cluster" "read_replica":
│   23: resource "azurerm_cosmosdb_postgresql_cluster" "read_replica" {
│
│ creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request"
│ Message="The source cluster is not ready for restore yet."
╵

3.
╷
│ Error: creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request" Message="pointInTimeUTC cannot be before 2023-07-10T19:43:44Z. If you need to restore your cluster to a point within last 35 days, open a support request."
│
│   with azurerm_cosmosdb_postgresql_cluster.read_replica,
│   on main.tf line 23, in resource "azurerm_cosmosdb_postgresql_cluster" "read_replica":
│   23: resource "azurerm_cosmosdb_postgresql_cluster" "read_replica" {
│
│ creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request"
│ Message="pointInTimeUTC cannot be before 2023-07-10T19:43:44Z. If you need to restore your cluster to a point within last 35 days, open a support request."
╵

4.
╷
│ Error: creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request" Message="Restoring a new cluster from a source cluster in another location is not supported yet."
│
│   with azurerm_cosmosdb_postgresql_cluster.read_replica,
│   on main.tf line 23, in resource "azurerm_cosmosdb_postgresql_cluster" "read_replica":
│   23: resource "azurerm_cosmosdb_postgresql_cluster" "read_replica" {
│
│ creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request"
│ Message="Restoring a new cluster from a source cluster in another location is not supported yet."

5.
╷
│ Error: creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request" Message="pointInTimeUTC cannot be in the future."
│
│   with azurerm_cosmosdb_postgresql_cluster.read_replica,
│   on main.tf line 23, in resource "azurerm_cosmosdb_postgresql_cluster" "read_replica":
│   23: resource "azurerm_cosmosdb_postgresql_cluster" "read_replica" {
│
│ creating Server Groupsv 2 (Subscription: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
│ Resource Group Name: "peterli07102023"
│ Server Groupsv 2 Name: "peterlitest0710secondary"): performing Create: clusters.ClustersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="bad_request"
│ Message="pointInTimeUTC cannot be in the future."
╵
```

### Expected Behaviour

Read replica should be created for the CosmosDB Postgresql cluster.

I tried using the REST API instead and found it can work properly:
[cosmosdb_postgresql_cluster](https://learn.microsoft.com/en-us/rest/api/postgresqlhsc/clusters/create?tabs=Python#create-a-new-cluster-as-a-read-replica)

### Actual Behaviour

Failure with the various error messages mentioned in the debug output.

### Steps to Reproduce

terraform plan -out main.tfplan -var-file="./postgre.tfvars"
terraform apply "main.tfplan"

### Important Factoids

N/A

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. Seems it's a question not a bug. Hopes below tf config is helpful.

tf config:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-pshsc-test01"
  location = "westeurope"
}

resource "azurerm_cosmosdb_postgresql_cluster" "test" {
  name                            = "acctestclustertest01"
  resource_group_name             = azurerm_resource_group.test.name
  location                        = azurerm_resource_group.test.location
  administrator_login_password    = "A@Bc1DoR4!"
  coordinator_storage_quota_in_mb = 131072
  coordinator_vcore_count         = 2
  node_count                      = 0
}

// Wait at least 20 minutes until all related components are ready
resource "time_sleep" "wait_20_minutes" {
  depends_on = [azurerm_cosmosdb_postgresql_cluster.test]

  create_duration = "1200s"
}

resource "azurerm_cosmosdb_postgresql_cluster" "test2" {
  name                 = "acctesttclustertest01"
  resource_group_name  = azurerm_resource_group.test.name
  location             = azurerm_resource_group.test.location
  source_location      = azurerm_cosmosdb_postgresql_cluster.test.location
  source_resource_id   = azurerm_cosmosdb_postgresql_cluster.test.id
  point_in_time_in_utc = azurerm_cosmosdb_postgresql_cluster.test.earliest_restore_time

  administrator_login_password    = "A@Bc1DoR4!"
  coordinator_storage_quota_in_mb = 131072
  coordinator_vcore_count         = 2
  node_count                      = 0

  depends_on = [time_sleep.wait_20_minutes]
}
```

**Comment by leilifremont** [comment]
Hi @neil-yechenwei,

Thank you for the quick turn around.

I tried to run your code, but got this error:

```
╷
│ Error: expected "point_in_time_in_utc" to be a valid RFC3339 date, got "": parsing time "" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "2006"
│
│   with azurerm_cosmosdb_postgresql_cluster.test2,
│   on main.tf line 133, in resource "azurerm_cosmosdb_postgresql_cluster" "test2":
│  133:   point_in_time_in_utc = azurerm_cosmosdb_postgresql_cluster.test.earliest_restore_time
│
╵
```


Are you able to run these commands on your end and create a read replica eventually? I asked the question because the expected behavior is a read replica should be created, you should be able to see this read replica from the "Replicate data globally" section of Azure portal.

Do you know which API will this resource call underlying? 
From Azure REST API document, they listed it as two APIs (one is create new cluster, the other is to create a read replica):
1.
Create a new cluster as a point in time restore
https://learn.microsoft.com/en-us/rest/api/postgresqlhsc/clusters/create?tabs=HTTP#create-a-new-cluster-as-a-point-in-time-restore

2.
Create a new cluster as a read replica
https://learn.microsoft.com/en-us/rest/api/postgresqlhsc/clusters/create?tabs=HTTP#create-a-new-cluster-as-a-read-replica

According to Microsoft API example code (HTTP), only in the first API (Create a new cluster as a point in time restore), they need specify the parameter pointInTimeUTC. In the API create read replica, this parameter is not necessary.

From my test, I can use the terraform resource azurerm_cosmosdb_postgresql_cluster to create a CosmosDB postgresql cluster without specify the pointInTimeUTC. This will be a primary replica. Then for the read replica, I can't create it use terraform resource azurerm_cosmosdb_postgresql_cluster. But I can create a read replica using the 2nd REST API then I can see this read replica from the "Replicate data globally" section of Azure portal.

**Comment by ben-childs-docusign** [comment]
@neil-yechenwei , @rcskosir this is a bug from our POV not a question. Can you review the update above and take a second look?

**Comment by neil-yechenwei** [comment]
I assume this issue has been fixed in the latest provider version v3.66.0. Suggest upgrade your provider version to latest one and try below tf config again to create read replica. Thanks.

tf config:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-pshsc-test01"
  location = "westeurope"
}

resource "azurerm_cosmosdb_postgresql_cluster" "test" {
  name                            = "acctestclustertest01"
  resource_group_name             = azurerm_resource_group.test.name
  location                        = azurerm_resource_group.test.location
  administrator_login_password    = "A@Bc1DoR4!"
  coordinator_storage_quota_in_mb = 131072
  coordinator_vcore_count         = 2
  node_count                      = 0
}

// Wait at least 20 minutes until all related components are ready
resource "time_sleep" "wait_20_minutes" {
  depends_on = [azurerm_cosmosdb_postgresql_cluster.test]

  create_duration = "1200s"
}

resource "azurerm_cosmosdb_postgresql_cluster" "test2" {
  name                 = "acctesttclustertest01"
  resource_group_name  = azurerm_resource_group.test.name
  location             = azurerm_resource_group.test.location
  source_location      = azurerm_cosmosdb_postgresql_cluster.test.location
  source_resource_id   = azurerm_cosmosdb_postgresql_cluster.test.id

  administrator_login_password    = "A@Bc1DoR4!"
  coordinator_storage_quota_in_mb = 131072
  coordinator_vcore_count         = 2
  node_count                      = 0

  depends_on = [time_sleep.wait_20_minutes]
}
```

**Comment by leilifremont** [comment]
Hi @neil-yechenwei, 
I tested using the latest release and I can create a read replica now. 

Now I have a few questions about the "time_sleep.wait_20_minutes" resource in the sample code,
1. Is there any specific consideration setting the waiting time to 20 minutes?
2. If we only UPDATE the primary cluster, I believe it may take much less time than creation. Is there a way to skip the 20-minute waiting period in such cases?
3. Is there a way to wait for the exact time that the primary cluster operation is complete?

Thanks!

**Comment by neil-yechenwei** [comment]
It's by service AP design. Suggest raise an ask on https://github.com/Azure/azure-rest-api-specs. I assume service team could help on this.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of v3.66.0. As such, I am going to mark this issue as closed. We encourage you to raise an ask on [https://github.com/Azure/azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs) with the questions above.
